src/model_io/urdf/src/URDFModelExport.cpp

```diff
@@ -0,0 +1,295 @@
+#include <iDynTree/Core/SpatialInertia.h>
+#include <iDynTree/Core/Transform.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/Model/Traversal.h>
+#include <iDynTree/Model/FixedJoint.h>
+#include <iDynTree/Model/PrismaticJoint.h>
+#include <iDynTree/Model/RevoluteJoint.h>
+
+#include "URDFParsingUtils.h"
+
+
+#include <iomanip>
+#include <fstream>
+#include <string>
+
+#include <libxml/tree.h>
+#include <libxml/parser.h>
+
+namespace iDynTree
+{
+
+/**
+ * Add a <origin> URDF element to the specified parent element with the specified transform.
+ */
+bool exportTransform(const Transform &trans, xmlNodePtr parent_element)
+{
+    bool ok = true;
+    xmlNodePtr origin = xmlNewChild(parent_element, NULL, BAD_CAST "origin", NULL);
+    std::string pose_xyz_str;
+    ok = ok && vectorToString(trans.getPosition(), pose_xyz_str);
+    std::string pose_rpy_str;
+    Vector3 pose_rpy = trans.getRotation().asRPY();
+    ok = ok && vectorToString(pose_rpy, pose_rpy_str);
+    xmlNewProp(origin, BAD_CAST "xyz", BAD_CAST pose_xyz_str.c_str());
+    xmlNewProp(origin, BAD_CAST "rpy", BAD_CAST pose_rpy_str.c_str());
+    return ok;
+}
+
+/**
+ * Add a <inertial> URDF element to the specified parent element with the specified 6D inertia.
+ */
+bool exportInertial(const SpatialInertia &i, xmlNodePtr parent_element)
+{
+    bool ok=true;
+    std::string bufStr;
+    xmlNodePtr inertial = xmlNewChild(parent_element, NULL, BAD_CAST "inertial", NULL);
+
+
+    xmlNodePtr mass_xml = xmlNewChild(inertial, NULL, BAD_CAST "mass", NULL);
+    ok = ok && doubleToStringWithClassicLocale(i.getMass(), bufStr);
+    xmlNewProp(mass_xml, BAD_CAST "value", BAD_CAST bufStr.c_str());
+
+    ok = ok && exportTransform(Transform(Rotation::Identity(), i.getCenterOfMass()), inertial);
+
+    xmlNodePtr inertia_xml = xmlNewChild(inertial, NULL, BAD_CAST "inertia", NULL);
+    RotationalInertiaRaw rotInertia = i.getRotationalInertiaWrtCenterOfMass();
+    ok = ok && doubleToStringWithClassicLocale(rotInertia(0, 0), bufStr);
+    xmlNewProp(inertia_xml, BAD_CAST "ixx", BAD_CAST bufStr.c_str());
+    ok = ok && doubleToStringWithClassicLocale(rotInertia(0, 1), bufStr);
+    xmlNewProp(inertia_xml, BAD_CAST "ixy", BAD_CAST bufStr.c_str());
+    ok = ok && doubleToStringWithClassicLocale(rotInertia(0, 2), bufStr);
+    xmlNewProp(inertia_xml, BAD_CAST "ixz", BAD_CAST bufStr.c_str());
+    ok = ok && doubleToStringWithClassicLocale(rotInertia(1, 1), bufStr);
+    xmlNewProp(inertia_xml, BAD_CAST "iyy", BAD_CAST bufStr.c_str());
+    ok = ok && doubleToStringWithClassicLocale(rotInertia(1, 2), bufStr);
+    xmlNewProp(inertia_xml, BAD_CAST "iyz", BAD_CAST bufStr.c_str());
+    ok = ok && doubleToStringWithClassicLocale(rotInertia(2, 2), bufStr);
+    xmlNewProp(inertia_xml, BAD_CAST "izz", BAD_CAST bufStr.c_str());
+
+    return ok;
+}
+
+/**
+ * Add a <link> URDF element to the specified parent element with the specified link info.
+ */
```

Comment:
Add the limitations that the current version does not support collisions and visuals.

Reply:
Already documented in https://github.com/robotology/idyntree/pull/554/files#diff-05b37777f3462332b9cce4040d3f16deR64 , will add it there.